snuba/datasets/errors_replacer.py

```diff
@@ -667,14 +665,126 @@ def process_unmerge(
         "previous_group_id": message["previous_group_id"],
         "project_id": message["project_id"],
         "timestamp": timestamp.strftime(DATETIME_FORMAT),
+        "hashes": _convert_hash_list(hashes, state_name),
     }
 
+    query_time_flags = (NEEDS_FINAL, message["project_id"])
+
+    return LegacyReplacement(
+        count_query_template, insert_query_template, query_args, query_time_flags
+    )
+
+
+def _convert_hash_list(
+    hashes: Sequence[str], state_name: ReplacerState, convert_types: bool = False
+) -> str:
     if state_name == ReplacerState.ERRORS:
-        query_args["hashes"] = ", ".join(
-            ["'%s'" % str(uuid.UUID(_hashify(h))) for h in hashes]
-        )
+        if convert_types:
+            return ", ".join(
+                "toUUID('%s')" % str(uuid.UUID(_hashify(h))) for h in hashes
+            )
```

Comment:
Why do we need to pass every attribute to the `toUUID` function ?

```
SELECT *
FROM errors_local
WHERE hasAny(hierarchical_hashes, ['986dd92c-be7f-11eb-b5fe-acde48001122', '98771c94-be7f-11eb-af7c-acde48001122'])

Ok.
```
just passing the UUIDS in quotes should be enough.

Also the call to `_hashify` should not be there. It generates an md5 if the hash is not already a valid one. Which would not be a UUID, so if we are not in the case it does nothing, then for errors the result would be broken anyway.
I would leave it there only when state_name is not ERRORS.

Reply:
we call hashify when inserting into the errors table like we do for primary_hash, and md5 is valid input for UUID (primary_hash itself is the output of `md5(stacktrace)`)

I believe I ran that query and ran into issues because `hasAny` got arrays of different types and did not cast by itself

Reply:
Ah yes, @fpacifici, the query will not error but silently return wrong results (always empty queryset)